clients/drcachesim/tests/scheduler_launcher.cpp

```diff
@@ -0,0 +1,201 @@
+    for (scheduler_t::stream_status_t status = stream->next_record(record);
+         status != scheduler_t::STATUS_EOF; status = stream->next_record(record)) {
+        if (status == scheduler_t::STATUS_WAIT) {
+            std::this_thread::yield();
+            continue;
+        }
+        if (status != scheduler_t::STATUS_OK)
+            FATAL_ERROR("scheduler failed to advance: %d", status);
+        if (thread_sequence.empty())
+            thread_sequence.push_back(record.instr.tid);
+        if (record.instr.tid != prev_tid) {
+            thread_sequence.push_back(record.instr.tid);
+            if (op_verbose.get_value() > 1) {
+                std::ostringstream line;
+                line
+                    << "Core #" << ordinal << " @" << stream->get_record_ordinal()
+                    << " refs, " << stream->get_instruction_ordinal() << " instrs: input "
+                    << stream->get_input_stream_ordinal() << " @"
+                    << scheduler
+                           .get_input_stream_interface(stream->get_input_stream_ordinal())
+                           ->get_record_ordinal()
+                    << " refs, "
+                    << scheduler
+                           .get_input_stream_interface(stream->get_input_stream_ordinal())
+                           ->get_instruction_ordinal()
+                    << " instrs == thread " << record.instr.tid << "\n";
+                std::cerr << line.str();
+            }
+            prev_tid = record.instr.tid;
+        }
+    }
+}
+
+} // namespace
+
+int
+_tmain(int argc, const TCHAR *targv[])
+{
+    // Convert to UTF-8 if necessary
+    char **argv;
+    drfront_status_t sc = drfront_convert_args(targv, &argv, argc);
+    if (sc != DRFRONT_SUCCESS)
+        FATAL_ERROR("Failed to process args: %d", sc);
+
+    std::string parse_err;
+    if (!droption_parser_t::parse_argv(DROPTION_SCOPE_FRONTEND, argc, (const char **)argv,
+                                       &parse_err, NULL) ||
+        op_trace_dir.get_value().empty()) {
+        FATAL_ERROR("Usage error: %s\nUsage:\n%s", parse_err.c_str(),
+                    droption_parser_t::usage_short(DROPTION_SCOPE_ALL).c_str());
+    }
+
+    scheduler_t scheduler;
+    std::vector<scheduler_t::input_workload_t> sched_inputs;
+    sched_inputs.emplace_back(op_trace_dir.get_value());
+    scheduler_t::scheduler_options_t sched_ops(
+        scheduler_t::MAP_TO_ANY_OUTPUT, scheduler_t::DEPENDENCY_IGNORE,
+        scheduler_t::SCHEDULER_DEFAULTS, op_verbose.get_value());
+    sched_ops.quantum_duration = op_sched_quantum.get_value();
+#ifdef HAS_ZLIB
+    std::unique_ptr<zipfile_ostream_t> record_zip;
+    std::unique_ptr<zipfile_istream_t> replay_zip;
+    if (!op_record_file.get_value().empty()) {
+        record_zip.reset(new zipfile_ostream_t(op_record_file.get_value()));
+        sched_ops.schedule_record_ostream = record_zip.get();
+    }
+    if (!op_replay_file.get_value().empty()) {
```

Comment:
else if? Because I suppose both cannot be set at the same time?

Reply:
Done.